### PR TITLE
Fix nonce issue for replay attack

### PR DESCRIPTION
**Pull request by araskachoi**
Closes: __#687__ #686

## Description

Checks the nonce and requires the nonce to be the same as the account sequence. 
before, the sequence needed to be greater than that of the nonce.

this fix will also throw an error when using an invalid nonce.
before, eth_sendTransaction would return a tx receipt even if the tx was rejected on the chain.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by summerpro — APPROVED** [review]

File: rpc/namespaces/eth/api.go [hunk 1]
```diff
@@ -1000,16 +1000,16 @@ func (api *PublicEthereumAPI) generateFromArgs(args rpctypes.SendTxArgs) (*evmty
...
+	// get the nonce from the account retriever and the pending transactions
+	nonce, err = api.accountNonce(api.clientCtx, args.From, true)
 	if err != nil {
 		return nil, err
 	}
+	if args.Nonce != nil {
+		if nonce != (uint64)(*args.Nonce) {
+			return nil, fmt.Errorf(fmt.Sprintf("invalid nonce; got %d, expected %d", (uint64)(*args.Nonce), nonce))
+		}
+	}
```
Comment (fedekunze) [inline rpc/namespaces/eth/api.go:1012]: nice, thanks for adding this client side validation!

**Review by fedekunze — APPROVED** [review]
ACK. Can you add a bug fix changelog entry under the Unreleased section?

**Comment by araskachoi** [comment]
> ACK. Can you add a bug fix changelog entry under the Unreleased section?

@fedekunze can you check the changelog once more before merging? Thanks!

**Comment by summerpro** [comment]
@araskachoi this pr cannot close #687 , but can close #686

**Comment by fedekunze** [comment]
> @araskachoi this pr cannot close #687 , but can close #686

@summerpro is right

**Comment by araskachoi** [comment]
> @araskachoi this pr cannot close #687 , but can close #686

agreed,

**Pull request merged**